### PR TITLE
Chore: Upgrade typescript to 4.1

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Bumps typescript to version 4.1

### CODE REVIEW

File: packages/grafana-data/src/field/fieldDisplay.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ export const getFieldDisplayValues = (options: GetFieldDisplayValuesOptions): Fi
             });
 
             values.push({
-              name,
+              name: '',
```
Comment (kaydelaney) [inline packages/grafana-data/src/field/fieldDisplay.ts:163]: As far as I understand, previously this code incorrectly referenced the (now deprecated) global [name](https://developer.mozilla.org/en-US/docs/Web/API/Window/name) property. Here I've replaced it with an empty string to satisfy the interface, but there might be a more appropriate value.

File: packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ const DemoBox: FC<DemoBoxProps> = ({ bg, border, children }) => {
           padding-bottom: 16px;
         `}
       >
-        {name}
+        {window.name}
```
Comment (dprokop) [inline packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx:34]: This looks funky to me. I guess this should be a `name` property maybe, but I cannot see anything like that in the props interface.

**Review by dprokop — APPROVED** [review]
Looks good @kaydelaney !

File: packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx (hunk 2 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx:34]: Yeah, it's just another instance of what I mentioned in my other comment. `{name}` was actually referencing the global `name` property (which is the same as `window.name`). I assume this wasn't the intended behavior, but I changed it to `window.name` to satisfy the type checker as well as at least not _changing_ the behavior, even if it wasn't actually meant. Maybe someone else can clarify what the original intention was. @torkelo ?

File: packages/grafana-data/src/field/fieldDisplay.ts (hunk 1 above)
Comment (kaydelaney) [reply packages/grafana-data/src/field/fieldDisplay.ts:163]: Yep

File: packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx (hunk 2 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx:34]: You can remove this whole div, not used, just results in an empty div https://developers.grafana.com/ui/latest/index.html?path=/story/docs-overview-themecolors--theme-colors-demo

**Review by torkelo — APPROVED** [review]

File: packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx (hunk 2 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/ThemeColors/ThemeColors.tsx:34]: Do we want to keep the `16px` padding?

**Pull request merged**